src/components/Admin/Approval/Approval.js

```diff
@@ -78,71 +78,83 @@ class ImageApproval extends Component {
-      this.setState({
-        storeList: Object.values({
-          ...storeList
-        })
-      });
-      this.getImages().then(() => this.setState({loading: false}));
-    });
+    Promise.all([
+      this.dataController.getStoreList().then(storeList => {
+        this.setState({
+          storeList
+        });
+      }),
+      this.getImages().then(() => {
+        this.displayNextImage();
+      })
+    ]).then(() => this.setState({loading: false}));
   }
 
   render() {
     if (this.state.loading) return <div>Loading...</div>;
     if (this.state.images === [])
       return <div>No untrusted images available</div>;
 
+    const {image} = this.state;
     return (
       <div className="preview">
         <div>
           <button className="button button-admin" onClick={this.back}>
             &laquo; Back
           </button>
         </div>
-        {this.state.images.map((image, index) => {
-          return (
-            <div key={image.id} className="preview-pane">
+        {image ? (
+          <div key={image.id} className="preview-pane">
+            <h2>
+              {this.state.storeList[image.label] !== undefined &&
+                this.state.storeList[image.label].name}
+            </h2>
+            <div>
               <img style={{maxHeight: 224}} src={image.url} alt="" />
-              <ItemSelector
-                item={image.label}
-                items={this.state.storeList}
-                setItem={cat => {
-                  this.changeCategory(image.id, cat);
-                  this.setState(prevState => {
-                    prevState.images[index].label = cat;
-                    return prevState;
-                  });
-                }}
-              />
-              <div>
-                <button
-                  className="button button-admin"
-                  data-id={image.id}
-                  data-index={index}
-                  data-label={image.label}
-                  onClick={this.trustImage}>
-                  Trust
-                </button>
-                <button
-                  className="button button-admin"
-                  data-id={image.id}
-                  data-index={index}
-                  onClick={this.setAsUnknown}>
-                  Unknown
-                </button>
-                <button
-                  className="button button-admin"
-                  data-id={image.id}
-                  data-index={index}
-                  data-label={image.label}
-                  onClick={this.deleteImage}>
-                  Delete
-                </button>
-              </div>
             </div>
-          );
-        })}
+            <ItemSelector
+              item={image.label}
+              items={Object.values(this.state.storeList)}
+              setItem={cat => {
+                this.changeCategory(image.id, cat);
+                this.setState(prevState => {
+                  prevState.image.label = cat;
```

Comment:
Could avoid mutation, slightly more verbose example

```js
this.setState(prevState => ({
  image: {
    ...prevState.images,
    label: cat
  }
})
```

Reply:
That logic doesn't quite do the same thing. Would something like this work even though `prevState.image` already has a label property?

```
this.setState(prevState => {
    return {image: {label: cat, ...prevState.image}};
});
```